### ISSUE TITLE
Tech Insights | Graph Contrastive Learning.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#60]
Tech Insights. [Graph contrastive learning](https://medium.com/@TechInsight/graph-contrastive-learning-b82dbef44f23).
Zhu Y, Xu Y, Liu Q, et al. [An empirical study of graph contrastive learning](https://arxiv.org/pdf/2109.01116.pdf).

**Comment by NorbertZheng** [comment]
## Overview
Getting high quality labeled dataset at scale for graph-related problems is often expensive. Graph neural networks tend to overfit small training data sets and fail to learn reusable, task-invariant knowledge.

Self-supervised learning has been hugely successful in multiple ML areas and has improved label efficiency. These techniques obtain supervisory signals from the unlabelled data by utilising the data’s underlying structure.

![image](https://user-images.githubusercontent.com/37235023/217138178-3feb778d-c9c8-4fbf-9cbe-f149aaada340.png)

**Comment by NorbertZheng** [comment]
## Graph Contrastive Learning
The goal of graph contrastive learning is to learn a low-dimensional representation to encode the graph’s structural and attribute information.

The contrastive learning paradigm tries to maximize the agreement between the latent representations under scholastic data augmentation. Essentially, it tries to distinguish between a pair of representations from the two augmentations of the same sample (positives) apart from the $k$ pairs of representation from the other (negative) samples.

![image](https://user-images.githubusercontent.com/37235023/217138354-170a65ad-85a1-4ab6-b07d-86274e7ba2aa.png)
[Chen et.al, 2020] A Simple Framework for Contrastive Learning for Visual Representation, in ICML 2020.

**Comment by NorbertZheng** [comment]
## Experiments
The authors have explored multiple design dimensions for optimizing performance in this paper. These four dimensions as listed as follows:
 - Data augmentation
 - Contrasting Modes
 - Contrastive objectives
 - Negative mining strategies

**Data augmentation**: Topology augmentation, feature augmentation, or both create augmentation as an input to the model. Topology augmentation includes graph structural changes like edge adding, edge removal, etc.
![image](https://user-images.githubusercontent.com/37235023/217138627-df7c4db9-7387-48b5-8f98-83d7d24ae3f5.png)

**Contrasting Modes**: This method generates different types of positives and negatives for the model. The same-scale contrastive ways create samples for each node (local-local) or the entire graph (global-global). While the cross-scale contrasting generates samples between the node and the whole graph, i.e. (local-global).
![image](https://user-images.githubusercontent.com/37235023/217138838-8238278c-73ac-4c3a-a984-c42c1cd014b9.png)


**Contrasting Objective**: The following methods utilize negative samples:
 - Information Noice Contrastive Estimation (InfoNCE),
 - Jensen-Shannon Divergence (JSD),
 - Triplet Margin loss (TM).

Moreover, they also analyze the following objectives that eschew the need for explicit negative samples:
 - the Bootstrapping Latent loss (BL),
 - Barlow Twins (BT) loss,
 - VICReg loss.

![image](https://user-images.githubusercontent.com/37235023/217139123-039177f7-1a5e-4f13-900b-1f0c55a97c3e.png)

**Negative mining**: Recent work argues that CL algorithms benefit from hard, negative samples (i.e. samples challenging to distinguish from an anchor instance). In this work, they have considered he following four negative mining strategies:
 - Hard Negative Mixing (HNM),
 - Debiased Contrastive Learning (DCL),
 - Hardness-Biased Negative Mining (HBNM),
 - Conditional Negative Mining (CNM).

**Datasets and tasks**: The models are first trained unsupervised, and the resulting embeddings are fed into a linear classifier to fit the labelled data. Model accuracy is reported for a different experiment, which has several combinations of the augmentation, mode, mining technique, and objective.
![image](https://user-images.githubusercontent.com/37235023/217139349-69f0690f-b08b-45ee-a154-743db3858795.png)

**Comment by NorbertZheng** [comment]
## Observations
 - Topology augmentation greatly affects model performance. Augmentation functions that produce sparser graphs generally lead to better performance.
   ![image](https://user-images.githubusercontent.com/37235023/217139489-16f7320a-dda7-45e9-8627-9bda865f7a71.png)
 - Feature augmentations bring additional benefits to GCL. Compositional augmentations at both structure and attribute level benefit GCL most.
   ![image](https://user-images.githubusercontent.com/37235023/217139576-746677ca-83ef-4fce-a33e-e73087d65cb0.png)
 - Same-scale contrasting generally performs better. Downstream tasks of different granularities favor different contrasting modes.
   ![image](https://user-images.githubusercontent.com/37235023/217139762-db75b826-9fb6-435a-a17b-55b499156f48.png)
 - Among negative-sample-based objectives, the use of InfoNCE objective leads to consistent improvements across all settings.
   ![image](https://user-images.githubusercontent.com/37235023/217139850-721f9ebc-df36-4a76-9690-26fd8e79398d.png)
 - Bootstrapping Latent and Barlow Twins losses obtain promising performance on par with their negative-sample-based counterparts yet reduce the computational burden without explicit negative samples.
   ![image](https://user-images.githubusercontent.com/37235023/217139929-4dd5d7a1-e8b0-4519-b1a3-6dc87f33b354.png)
 - Existing negative mining techniques based on calculating embedding similarities bring limited benefit to GCL.
   ![image](https://user-images.githubusercontent.com/37235023/217140062-87c70988-8073-4490-9e45-997ccac79107.png)

**Comment by NorbertZheng** [comment]
## References
 - [Graph Neural Networks: Foundations, Frontiers, and Applications](https://amzn.to/3thdEmE).
 - [Graph Neural Networks in Action](https://amzn.to/3UEJ95A).
 - [Graph Contrastive Learning](https://arxiv.org/pdf/2109.01116.pdf).
